packages/editor/src/panels/assets/hooks.tsx

```diff
@@ -140,9 +140,22 @@ export const AssetsQueryProvider = ({ children }: { children: ReactNode }) => {
     return () => abortSignal()
   }, [])
 
+  function convertToHierarchy(obj: Record<string, any>, depth = 0, parentPath = ''): AssetCategoryNode[] {
+    return Object.entries(obj).map(([key, value]) => {
+      const currentPath = parentPath ? `${parentPath}/${key}` : key
+
+      return {
+        name: key,
+        path: currentPath,
+        depth,
+        children: convertToHierarchy(value, depth + 1, currentPath)
+      }
+    })
+  }
+
   useEffect(() => {
-    categories.set(mapCategoriesHelper(expandedCategories.value))
-  }, [expandedCategories])
+    categories.set(convertToHierarchy(AssetsPanelCategories.initial))
+  }, [])
```

Comment:
in this case, do not use state for `categories`
just pass it as constant variable
you can initialize this constant variable outside the react function to avoid `useMemo`